### PR TITLE
Api set model hook retries

### PR DESCRIPTION
**Pull request by blackboxsw**
Add new Juju2API.setModelConfig method to allow for enabling and disabling model-config settings for things like Juju's automatically-retry-hooks feature. This will allow openstack autopilot deployments to be strictly honor hook failures during cloud deployments.

I avoided extending Juju2Cli as we generally should only use that cli for things which cannot be performed with API calls.

This pull request also adds:
 - travis integration
 - separates integration tests from unit tests  as integration tests rely on fake-juju
 - new integration-test make target

### CODE REVIEW

File: txjuju/tests/test_api.py [hunk 1]
```diff
@@ -934,6 +934,42 @@ def test_modelInfo_bad_result(self):
         self.assertEqual("malformed result {}", err.value.error)
         self.assertEqual("", err.value.code)
 
+    def test_setModelConfig_good_result(self):
+        """
+        The setModelConfig method sends an 'ModelConfig' request and
```
Comment (simpoir) [inline txjuju/tests/test_api.py]: s/an/a/

File: txjuju/tests/test_api.py [hunk 2]
```diff
@@ -934,6 +934,42 @@ def test_modelInfo_bad_result(self):
...
+        """
+        deferred = self.client.setModelConfig(
+            "automatically-retry-hooks", False)
+        params = {"config": {"automatically-retry-hooks": False}}
+        self.assertEqual("ModelConfig", self.backend.lastType)
+        self.assertEqual("ModelSet", self.backend.lastRequest)
+        self.assertEqual(1, self.backend.lastVersion)
+        self.assertEqual(params, self.backend.lastParams)
+        self.backend.response({})
+        self.assertIsNone(self.successResultOf(deferred))
```
Comment (simpoir) [inline txjuju/tests/test_api.py]: Shouldn't this result not be None according to the test docstring?

Comment (blackboxsw) [inline txjuju/tests/test_api.py]: While the API returns an empty dict {} in response, our _parseSetModelConfigResult essentially discards anything other than error messages and returns None.

File: txjuju/tests/test_api.py (hunk 1 above)
Comment (blackboxsw) [inline txjuju/tests/test_api.py]: fixed

**Pull request merged**